homework/index.js

```diff
@@ -30,15 +33,189 @@
+            href: item.html_url,
+            target: '_blank',
+          });
+          createAndAppend('div', contributorDiv, {
+            id: `${item.login}Badge`,
+            class: 'cont-badge',
+            text: item.contributions,
+          });
+        });
+      },
+      err => {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      },
+    );
+  }
+
+  function updateContributorSection(selected) {
+    const contributorsDiv = document.getElementById('contsDiv');
+    contributorsDiv.innerHTML = '';
+    createAndAppend('p', contributorsDiv, { id: 'cont-header', text: 'Contributors' });
+    const contributorsLists = createAndAppend('ul', contributorsDiv, { id: 'conts-list' });
+
+    fetchJSON(`https://api.github.com/repos/HackYourFuture/${selected}/contributors`).then(
+      data => {
+        data.forEach(item => {
+          const contributorListItem = createAndAppend('li', contributorsLists, {
+            id: `${item.login}ListItem`,
+            class: 'cont-li',
+          });
+
+          createAndAppend('img', contributorListItem, {
+            id: `${item.login}Img`,
+            class: 'cont-avatar',
+            src: item.avatar_url,
+          });
+
+          const contributorDiv = createAndAppend('div', contributorListItem, {
+            id: `${item.login}Div`,
+          });
+
+          createAndAppend('a', contributorDiv, {
+            id: `${item.login}Name`,
+            class: 'cont-name',
+            text: item.login,
+            href: item.html_url,
+            target: '_blank',
+          });
+          createAndAppend('div', contributorDiv, {
+            id: `${item.login}Badge`,
+            class: 'cont-badge',
+            text: item.contributions,
+          });
+        });
+      },
+      err => {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      },
+    );
+  }
+
+  function updateInfoSection(selected, data) {
+    const selectedData = getSelectedData(selected, data);
+
+    const repoName = document.getElementById('repoName');
+    repoName.innerText = '';
+    repoName.innerText = `Repository: ${selectedData.name}`;
+    repoName.href = selectedData.html_url;
+    repoName.target = '_blank';
+
+    const desc = document.getElementById('desc');
+    desc.innerText = '';
+    desc.innerText = `Description: ${selectedData.description}`;
+
+    const forks = document.getElementById('forks');
+    forks.innerText = '';
+    forks.innerText = `Forks: ${selectedData.forks}`;
+
+    const updatedAt = document.getElementById('updated_at');
+    updatedAt.innerText = '';
+    updatedAt.innerText = `Updated: ${selectedData.updated_at}`;
+  }
+
+  function main(url) {
+    fetchJSON(url).then(
+      data => {
+        const repoName = data.map(repo => repo.name).sort((a, b) => a.localeCompare(b));
+        generateSelections(repoName);
+
+        const selected = document.getElementById('repositories');
```

Comment:
This might have been missed earlier but I think you can combine the two statements together in a single line i.e instead of calling `generateSelections()` and then trying to fetch `repositories` element created by it, you can simply update your function to return the newly created element and store that in the return value.

```
function generateSelections(repos) {
  // code remains the same
  return selectionElem;
}

// Inside main()
const selected = generateSelections(repoName);
```

Reply:
I think you missed addressing this point in your updates. I am approving the PR right now but please do have a look. It saves you one operation of adding an element and then finding it using `document.getElementById` in the next step. 🙂